Leo annual subscription on Android

SergeyZhukovsky opened the pull request:
<!-- Add brave-browser issue below that this PR will resolve -->
Resolves https://github.com/brave/brave-browser/issues/41008

<!-- CI-related labels that can be applied to this PR:
* CI/run-audit-deps (1) - check for known npm/cargo vulnerabilities (audit_deps)
* CI/run-network-audit (1) - run network-audit
* CI/run-upstream-tests - run Chromium unit and browser tests on Linux and Windows (otherwise only on Linux)
* CI/run-linux-arm64, CI/run-macos-x64, CI/run-windows-arm64, CI/run-windows-x86 - run builds that would otherwise be skipped
* CI/skip - do not run CI builds (except noplatform)
* CI/skip-linux-x64, CI/skip-android, CI/skip-macos-arm64, CI/skip-ios, CI/skip-windows-x64 - skip CI builds for specific platforms
* CI/skip-upstream-tests - do not run Chromium unit, or browser tests (otherwise only on Linux)
* CI/skip-all-linters - do not run presubmit and lint checks
* CI/storybook-url (1) - deploy storybook and provide a unique URL for each build

(1) applied automatically when some files are changed (see: https://github.com/brave/brave-core/blob/master/.github/labeler.yml)
-->

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

1. Make sure there is a trial period when purchasing Leo Yearly and monthly.
2. Make sure it is still premium after trial period ends.

deeppandya reviewed (commented):
Overall LGTM. We need to update style for new design based on figma. i.e. instead of using textSize, we should style from figma to make things uniform.

SergeyZhukovsky commented:
> Overall LGTM. We need to update style for new design based on figma. i.e. instead of using textSize, we should style from figma to make things uniform.

@deeppandya I updated text views I added except the `BEST VALUE`(we will need to correct a script to generate it right).

mkarolin reviewed (approved):
`strings`++

deeppandya reviewed (approved):
LGTM

brave-builds commented:
Released in [v1.73.13](https://github.com/brave/brave-browser/releases/tag/v1.73.13)

hffvld commented:
Verified on `Pixel 7` using version(s):
```
Device/OS: Pixel 7 / panther_beta-user 15 AP41.240925.009 release-keys
Brave build: 1.73.75 
Chromium: 131.0.6778.24 (Official Build) canary (64-bit) 
```
---------------------------

`Fresh account: no previous purchases of Leo subscription`

**STEPS:**
1. Launch Brave
2. Hamburger menu > Settings
3. Leo > Go Premium > *Verify*
4. Select `One year` > Try 7 days free > *Verify*
5. Go back and select `Monthly` > Try 7 days free > *Verify*

**ACTUAL RESULTS:**
- Verified that the `Leo` paywall has 2 subscription options `One year` and `Monthly`.
- Verified that the `7-day free trial` is available for the account that didn't purchase a `Leo` subscription previously and is purchasing a `One year` subscription now.
- Verified that the `7-day free trial` is available for the account that didn't purchase a `Leo` subscription previously and is purchasing a `Monthly` subscription now.
- Verified that the `yearly` Leo subscription can be purchased successfully.

---------------------------

|1|2|3|4|5|
|:-:|:-:|:-:|:-:|:-:|
| ![1](https://github.com/user-attachments/assets/65835432-9e40-4a0f-a25d-d67702c98a85) | ![2](https://github.com/user-attachments/assets/9d9608f8-7211-429d-89bf-229182409aa2) | ![3](https://github.com/user-attachments/assets/00059b5a-8bce-4905-a24a-faf4329c9f90) | ![4](https://github.com/user-attachments/assets/0c67fbd5-2403-4347-b711-b4ac987a9104) | ![5](https://github.com/user-attachments/assets/607a08fb-6c1f-4dcb-b7b5-b107561e5c12) |

deeppandya commented:
Verified on `Pixel fold` using version(s):
```
Device/OS: Pixel fold / Android 15 
Brave build: 1.73.75 
Chromium: 131.0.6778.24 (Official Build) canary (64-bit) 
```
---------------------------

`Fresh account: no previous purchases of Leo subscription`
**Steps:**
1. Launch Brave
2. Hamburger menu > Settings
3. Leo > Go Premium
4. Select `Monthly plan` > Try 7 days free > *Verify*
5. Cancel the subscription from play console and wait for the subscription duration
6. Go back and select `Yearly` > Try 7 days free > *Verify*

**ACTUAL RESULTS:**
- Verified that the `Leo` paywall has 2 subscription options `One year` and `Monthly`.
- Verified that the `7-day free trial` is available for the account that didn't purchase a `Leo` subscription previously and is purchasing a `Monthly` subscription now.
- Verified that the `7-day free trial` is available for the account that didn't purchase a `Leo` subscription previously and is purchasing a `One year` subscription now.
- Verified that the `yearly` Leo subscription can be purchased successfully.
